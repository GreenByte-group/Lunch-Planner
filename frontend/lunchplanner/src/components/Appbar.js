import React from 'react';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import Button from 'material-ui/Button';
import IconButton from 'material-ui/IconButton';
import IconSearch from '@material-ui/icons/Search';
import { withStyles } from 'material-ui/styles';


import LunchMenu from "./LunchMenu";

const styles = {
    root: {
        flexGrow: 1,
    },
    flex: {
        flex: 1,
        fontFamily: "Work Sans",
        fontWeight: '600',
    },
    menuButton: {
        marginLeft: -12,
        marginRight: 20,
    },
    appbar: {
        position: 'fixed',
        boxShadow: 'none',
    }
};

function ButtonAppBar(props) {
    const { classes } = props;
    let titel = props.currentScreen;
    return (
        <div className={classes.root}>
            <AppBar className={classes.appbar}>
                <Toolbar>
                    <IconButton className={classes.menuButton} color="inherit" aria-label="Menu">
                        <LunchMenu open={false}/>
                    </IconButton>
                    <Typography variant="Events" color="inherit" className={classes.flex}>
                        {titel}
                    </Typography>
                    <Button color = "inherit">
                        <IconSearch/>
                    </Button>
                </Toolbar>
            </AppBar>
        </div>
    );
}

export default withStyles(styles)(ButtonAppBar);