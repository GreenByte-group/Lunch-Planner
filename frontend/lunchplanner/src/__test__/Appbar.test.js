import React from 'react';
import Appbar from '../components/Appbar';
import renderer from 'react-test-renderer';
import {MemoryRouter} from "react-router-dom";

it ("renders correctly", ()=>{
    const tree = renderer.create(
        <Appbar/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});