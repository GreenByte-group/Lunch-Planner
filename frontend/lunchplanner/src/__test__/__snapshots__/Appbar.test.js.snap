// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ButtonAppBar-root-1"
>
  <header
    className="MuiPaper-root-13 MuiPaper-elevation4-19 MuiAppBar-root-5 MuiAppBar-positionFixed-6 MuiAppBar-colorPrimary-11 mui-fixed ButtonAppBar-appbar-4"
  >
    <div
      className="MuiToolbar-root-40 MuiToolbar-gutters-41"
    >
      <div
        aria-label="Menu"
        className="ButtonAppBar-menuButton-3"
        color="inherit"
      >
        <div>
          <button
            className="MuiButtonBase-root-63 MuiButton-root-47"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-48"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-66"
                color={undefined}
                focusable="false"
                style={
                  Object {
                    "color": "white",
                  }
                }
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-72"
            />
          </button>
        </div>
      </div>
      <p
        className="MuiTypography-root-79 MuiTypography-body1-88 MuiTypography-colorInherit-98 ButtonAppBar-flex-2"
      />
      <div
        color="inherit"
      >
        <div>
          <button
            className="MuiButtonBase-root-63 MuiButton-root-47"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-48"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-66"
                color={undefined}
                focusable="false"
                style={
                  Object {
                    "color": "white",
                  }
                }
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-72"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;
